model/waterfall.go

```diff
@@ -48,6 +49,68 @@ type WaterfallOptions struct {
 	MaxOrder   int      `bson:"-" json:"-"`
 	MinOrder   int      `bson:"-" json:"-"`
 	Requesters []string `bson:"-" json:"-"`
+	Variants   []string `bson:"-" json:"-"`
+}
+
+// Older versions don't have their build display names saved in the version document.
+// For those missing it, look up their build details.
+// TODO DEVPROD-15118: This function can be removed after 7 February 2026, when the version TTL index applies and all versions include build display names.
+func getBuildDisplayNames(match bson.M) bson.M {
+	match[bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusDisplayNameKey)] = bson.M{"$exists": false}
+	return bson.M{
+		"$unionWith": bson.M{
+			"coll": VersionCollection,
+			"pipeline": []bson.M{
+				bson.M{"$match": match},
+				bson.M{"$sort": bson.M{VersionRevisionOrderNumberKey: -1}},
+				bson.M{"$limit": MaxWaterfallVersionLimit},
+				bson.M{
+					"$lookup": bson.M{
+						"from":         build.Collection,
+						"localField":   bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusIdKey),
+						"foreignField": build.IdKey,
+						"as":           VersionBuildVariantsKey,
+						"pipeline": []bson.M{
+							bson.M{
+								"$project": bson.M{
+									build.DisplayNameKey:    1,
```

Comment:
nit. some of these `bson.M`'s seem to be implied and redundant 

<img width="263" alt="Screenshot 2025-02-21 at 12 55 30 PM" src="https://github.com/user-attachments/assets/6c0e3eaf-202b-4991-91a5-354030252a1a" />

Reply:
They are pretty commonly used in the codebase for nested maps, @hadjri is there any best practice here?